### DISCUSSION TITLE
Constraints with summation

### DISCUSSION THREAD

**Discussion by sb1368** [yalmip/YALMIP#1428] [Optimization modelling] [answered]
Hi everybody,

I have the following constraint in my optimization model:

sp(i,c)-s(i,c)+sum(j,sub(j,i,c))= 0      for all i and c

sp,s, and sub are decision variables.

How can I write this constraint using YALMIP? I examine two following codes to know the answer of this question but I get errors (obj is not important):

clc
clear

I=40; J=40; C=15;
s=sdpvar(I,C);
sp=sdpvar(I,C);
sub=sdpvar(J,I,C);

sumsub=sum(sub,1)
sumsubp(:,:)=sumsub(1,:,:)

sizes=size(s)
sizesend=size(sp)
sizesumsubp=size(sumsubp)

con=[sp-s+sumsubp==0]
obj=sum(sp)-sum(sumsubp);
 sett=sdpsettings('solver','sedumi');
 optimize(con,obj,sett);
 Cost=value(obj)

Error using  +  (line 35)
Cannot add SDPVAR object and NDSDPVAR object

If I replace the above con with the below (By multiplying -1 ),

con=[-sp+s-sumsubp==0]

then the error is:
Error using  | 
NaN's cannot be converted to logicals.

Error in lmi/categorizeproblem>multipletermsInEquality (line 509)
    p = any(sum(B | B,2)-(B(:,1) == 0) > 1);

Error in lmi/categorizeproblem (line 163)
                if multipletermsInEquality(Fi)

Error in compileinterfacedata (line 298)
[ProblemClass,integer_variables,binary_variables,parametric_variables,uncertain_variables,semicont_variables,quad_info]
= categorizeproblem(F,logdetStruct,h,options.relax,parametric,evaluation_based,F_vars,exponential_cone);

Error in solvesdp (line 301)
[interfacedata,recoverdata,solver,diagnostic,F,Fremoved,ForiginalQuadratics] =
compileinterfacedata(F,[],logdetStruct,h,options,0,solving_parametric,solvers);

Error in solvesdp_multiple (line 27)
model = solvesdp(varargin{:});

Error in solvesdp (line 22)
        diagnostic = solvesdp_multiple(varargin{:});

Error in optimize (line 31)
[varargout{1:nargout}] = solvesdp(varargin{:});

Error in Untitled7 (line 19)
 optimize(con,obj,sett);

What is the best way to consider the mentioned constraint in an optimization modeling?

**Comment by johanlofberg** [comment, marked as answer]
Dimensions/class don't match. You probably want

`sumsubp = squeeze(sumsub(1,:,:))`

or 

`sumsubp = squeeze(sumsub)`

**Reply by sb1368** [reply]
Many thanks for your help. The frist one works for me.
